components/mediation/event-sink/org.wso2.micro.integrator.event.sink/src/main/java/org/wso2/micro/integrator/event/sink/EventSink.java

```diff
@@ -0,0 +1,179 @@
+
+        EventSink eventSink = new EventSink();
+
+        OMElement receiverUrl = eventSinkElement.getFirstChildWithName(EventSinkConstants.RECEIVER_URL_Q);
+        if (receiverUrl == null || "".equals(receiverUrl.getText())) {
+            throw new EventSinkException(
+                    EventSinkConstants.RECEIVER_URL_Q.getLocalPart() + " is missing in thrift endpoint config");
+        }
+        eventSink.setReceiverUrlSet(receiverUrl.getText());
+
+        OMElement authenticatorUrl = eventSinkElement.getFirstChildWithName(EventSinkConstants.AUTHENTICATOR_URL_Q);
+        if (authenticatorUrl != null) {
+            eventSink.setAuthenticationUrlSet(authenticatorUrl.getText());
+        }
+
+        OMElement userName = eventSinkElement.getFirstChildWithName(EventSinkConstants.USERNAME_Q);
+        if (userName == null || "".equals(userName.getText())) {
+            throw new EventSinkException(
+                    EventSinkConstants.USERNAME_Q.getLocalPart() + " is missing in thrift endpoint config");
+        }
+        eventSink.setUsername(userName.getText());
+
+        OMElement password = eventSinkElement.getFirstChildWithName(EventSinkConstants.PASSWORD_Q);
+        if (password == null || "".equals(password.getText())) {
+            throw new EventSinkException(
+                    EventSinkConstants.PASSWORD_Q.getLocalPart() + " attribute missing in thrift endpoint config");
+        }
+
+        try {
+            eventSink.setPassword(
+                    new String(CryptoUtil.getDefaultCryptoUtil().base64DecodeAndDecrypt(password.getText()),
+                            Charset.forName("UTF-8")));
+        } catch (CryptoException e) {
+            throw new EventSinkException("Failed to decrypt password", e);
+        }
+        eventSink.setName(name);
+
+        String authenticationUrlSet = null;
+        if (eventSink.getAuthenticationUrlSet().trim().length() > 0) {
+            authenticationUrlSet = eventSink.getAuthenticationUrlSet();
+        }
+        try {
+            eventSink.setDataPublisher(
+                    new DataPublisher(DataEndpointConstants.THRIFT_DATA_AGENT_TYPE, eventSink.getReceiverUrlSet(),
+                            authenticationUrlSet, eventSink.getUsername(),
+                            eventSink.getPassword()));
+        } catch (DataEndpointAgentConfigurationException | DataEndpointException | DataEndpointConfigurationException
+                | DataEndpointAuthenticationException | TransportException e) {
+            throw new EventSinkException("Error creating DataPublisher for the event sink ", e);
+        }
+        return eventSink;
+    }
+
+    /**
+     * Stop DataPublisher before removing the Event Sink.
+     *
+     * @param eventSink name of the Event Sink which is going to be removed
+     */
+    public static void stopDataPublisher(EventSink eventSink) throws EventSinkException {
+        if (eventSink != null && eventSink.getDataPublisher() != null) {
```

Comment:
Can the event sink be null? Also can there be an even sink without a data publisher?